sql/branch.sql

```diff
@@ -0,0 +1,24 @@
+BEGIN;
+    -- Tips that have been set to zero should have zero dues.
+
+    UPDATE payment_instructions SET due = 0 WHERE amount = 0 AND due != 0;
+
+    -- If a user has changed the tip amount at any point of time - we don't have a
+    -- way to properly repair the due, so we set it to zero.
+
+    UPDATE payment_instructions pi
+       SET due = 0
+     WHERE due > 9.41
+       AND (
+             SELECT COUNT(*)                    -- Tips with a different amount exist.
+               FROM payment_instructions pi2
+              WHERE pi.participant = pi2.participant
+                AND pi.team = pi2.team
+                AND amount != pi.amount
+           ) > 0;
+
+    -- For the rest of the payment_instructions, we set due to the highest multiple of the
+    -- tip amount that is lower than the minimum charge.
+
+    UPDATE payment_instructions SET due = floor(9.41/(amount)) * amount WHERE due > 9.41;
```

Comment:
> Hmm, this is starting to get tricky..
> Really I suppose we'd want to replay

Well, we've got [620](https://github.com/gratipay/gratipay.com/pull/3981#issuecomment-207423101) folks in this boat, at roughly $10 each, so call it $6,000 we'd be leaving on the table if we hard-reset those 620 to zero. Some—many? most?—of those folks will *never* fix their card and start giving again.

Raises the question of whether dues shouldn't expire after a while anyway. If I forget about Gratipay and come back a year later and re-up, I'd be surprised if I got hit with $10 that had been waiting there for a year. Should we expire dues after a 30 (60, 90) days?